### ISSUE TITLE
Whois V1.20 loops outputting the whois information over and over

### ISSUE THREAD

**Issue by BrianG61UK** [MicrosoftDocs/sysinternals#75]
When using whois from a cmd prompt in Windows 7 Ultimate 64 bit I find it often loops outputting the whois information over and over, only stopping then I use Ctrl/C.
e.g.

```
C:\Users\Brian G>whois namecheap.com

Whois v1.20 - Domain information lookup
Copyright (C) 2005-2017 Mark Russinovich
Sysinternals - www.sysinternals.com

Connecting to COM.whois-servers.net...

WHOIS Server: whois.enom.com
   Registrar URL: http://www.enom.com
   Updated Date: 2017-06-16T21:53:04Z
   Creation Date: 2000-08-11T16:15:25Z
   Registry Expiry Date: 2025-08-11T16:15:25Z
   Registrar: eNom, Inc.
   Registrar IANA ID: 48
   Registrar Abuse Contact Email:
   Registrar Abuse Contact Phone:
   Domain Status: clientTransferProhibited https://icann.org/epp#clientTransferProhibited
   Name Server: A1.VERISIGNDNS.COM
   Name Server: A2.VERISIGNDNS.COM
   Name Server: A3.VERISIGNDNS.COM
   DNSSEC: unsigned
   URL of the ICANN Whois Inaccuracy Complaint Form: https://www.icann.org/wicf/
>>> Last update of whois database: 2018-03-19T20:03:44Z <<<

For more information on Whois status codes, please visit https://icann.org/epp

NOTICE: The expiration date displayed in this record is the date the
registrar's sponsorship of the domain name registration in the registry is
currently set to expire. This date does not necessarily reflect the expiration
date of the domain name registrant's agreement with the sponsoring
registrar.  Users may consult the sponsoring registrar's Whois database to
view the registrar's reported date of expiration for this registration.

TERMS OF USE: You are not authorized to access or query our Whois
database through the use of electronic processes that are high-volume and
automated except as reasonably necessary to register domain names or
modify existing registrations; the Data in VeriSign Global Registry
Services' ("VeriSign") Whois database is provided by VeriSign for
information purposes only, and to assist persons in obtaining information
about or related to a domain name registration record. VeriSign does not
guarantee its accuracy. By submitting a Whois query, you agree to abide
by the following terms of use: You agree that you may use this Data only
for lawful purposes and that under no circumstances will you use this Data
to: (1) allow, enable, or otherwise support the transmission of mass
unsolicited, commercial advertising or solicitations via e-mail, telephone,
or facsimile; or (2) enable high volume, automated, electronic processes
that apply to VeriSign (or its computer systems). The compilation,
repackaging, dissemination or other use of this Data is expressly
prohibited without the prior written consent of VeriSign. You agree not to
use electronic processes that are automated and high-volume to access or
query the Whois database except as reasonably necessary to register
domain names or modify existing registrations. VeriSign reserves the right
to restrict your access to the Whois database in its sole discretion to ensure
operational stability.  VeriSign may restrict or terminate your access to the
Whois database for failure to abide by these terms of use. VeriSign
reserves the right to modify these terms at any time.

The Registry database contains ONLY .COM, .NET, .EDU domains and
Registrars.

Connecting to whois.enom.com...

WHOIS Server: whois.enom.com
Registrar URL: www.enom.com
Updated Date: 2017-01-13T05:52:52.00Z
Creation Date: 2000-08-11T12:15:25.00Z
Registrar Registration Expiration Date: 2025-08-11T16:15:00.00Z
Registrar: ENOM, INC.
Registrar IANA ID: 48
Reseller: NAMECHEAP, INC
Domain Status: clientTransferProhibited https://www.icann.org/epp#clientTransferProhibited
Registry Registrant ID:
Registrant Name: NAMECHEAP.COM NAMECHEAP.COM
Registrant Organization: NAMECHEAP, INC
Registrant Street: 4600 EAST WASHINGTON STREET, SUITE 305
Registrant City: PHOENIX
Registrant State/Province: AZ
Registrant Postal Code: 85034
Registrant Country: US
Registrant Phone: +1.6613102107
Registrant Phone Ext:
Registrant Fax: +1.6613102107
Registrant Fax Ext:
Registrant Email: SUPPORT@NAMECHEAP.COM
Registry Admin ID:
Admin Name: NAMECHEAP.COM NAMECHEAP.COM
Admin Organization: NAMECHEAP, INC
Admin Street: 4600 EAST WASHINGTON STREET, SUITE 305
Admin City: PHOENIX
Admin State/Province: AZ
Admin Postal Code: 85034
Admin Country: US
Admin Phone: +1.6613102107
Admin Phone Ext:
Admin Fax: +1.6613102107
Admin Fax Ext:
Admin Email: SUPPORT@NAMECHEAP.COM
Registry Tech ID:
Tech Name: NAMECHEAP.COM NAMECHEAP.COM
Tech Organization: NAMECHEAP, INC
Tech Street: 4600 EAST WASHINGTON STREET, SUITE 305
Tech City: PHOENIX
Tech State/Province: AZ
Tech Postal Code: 85034
Tech Country: US
Tech Phone: +1.6613102107
Tech Phone Ext:
Tech Fax: +1.6613102107
Tech Fax Ext:
Tech Email: SUPPORT@NAMECHEAP.COM
Name Server: A1.VERISIGNDNS.COM
Name Server: A2.VERISIGNDNS.COM
Name Server: A3.VERISIGNDNS.COM
DNSSEC: unSigned
Registrar Abuse Contact Email: abuse@enom.com
Registrar Abuse Contact Phone: +1.4252982646
URL of the ICANN WHOIS Data Problem Reporting System: http://wdprs.internic.net/
>>> Last update of WHOIS database: 2017-01-13T05:52:52.00Z <<<

For more information on Whois status codes, please visit https://icann.org/epp


The data in this whois database is provided to you for information
purposes only, that is, to assist you in obtaining information about or
related to a domain name registration record. We make this information
available "as is," and do not guarantee its accuracy. By submitting a
whois query, you agree that you will use this data only for lawful
purposes and that, under no circumstances will you use this data to: (1)
enable high volume, automated, electronic processes that stress or load
this whois database system providing you this information; or (2) allow,
enable, or otherwise support the transmission of mass unsolicited,
commercial advertising or solicitations via direct mail, electronic
mail, or by telephone. The compilation, repackaging, dissemination or
other use of this data is expressly prohibited without prior written
consent from us.

We reserve the right to modify these terms at any time. By submitting
this query, you agree to abide by these terms.
Version 6.3 4/3/2002

Get Noticed on the Internet!  Increase visibility for this domain name by listing it at www.whoisbusinesslistings.com
Connecting to whois.enom.com...

WHOIS Server: whois.enom.com
Registrar URL: www.enom.com
Updated Date: 2017-01-13T05:52:52.00Z
Creation Date: 2000-08-11T12:15:25.00Z
Registrar Registration Expiration Date: 2025-08-11T16:15:00.00Z
Registrar: ENOM, INC.
Registrar IANA ID: 48
Reseller: NAMECHEAP, INC
Domain Status: clientTransferProhibited https://www.icann.org/epp#clientTransferProhibited
Registry Registrant ID:
Registrant Name: NAMECHEAP.COM NAMECHEAP.COM
Registrant Organization: NAMECHEAP, INC
Registrant Street: 4600 EAST WASHINGTON STREET, SUITE 305
Registrant City: PHOENIX
Registrant State/Province: AZ
Registrant Postal Code: 85034
Registrant Country: US
Registrant Phone: +1.6613102107
Registrant Phone Ext:
Registrant Fax: +1.6613102107
Registrant Fax Ext:
Registrant Email: SUPPORT@NAMECHEAP.COM
Registry Admin ID:
Admin Name: NAMECHEAP.COM NAMECHEAP.COM
Admin Organization: NAMECHEAP, INC
Admin Street: 4600 EAST WASHINGTON STREET, SUITE 305
Admin City: PHOENIX
Admin State/Province: AZ
Admin Postal Code: 85034
Admin Country: US
Admin Phone: +1.6613102107
Admin Phone Ext:
Admin Fax: +1.6613102107
Admin Fax Ext:
Admin Email: SUPPORT@NAMECHEAP.COM
Registry Tech ID:
Tech Name: NAMECHEAP.COM NAMECHEAP.COM
Tech Organization: NAMECHEAP, INC
Tech Street: 4600 EAST WASHINGTON STREET, SUITE 305
Tech City: PHOENIX
Tech State/Province: AZ
Tech Postal Code: 85034
Tech Country: US
Tech Phone: +1.6613102107
Tech Phone Ext:
Tech Fax: +1.6613102107
Tech Fax Ext:
Tech Email: SUPPORT@NAMECHEAP.COM
Name Server: A1.VERISIGNDNS.COM
Name Server: A2.VERISIGNDNS.COM
Name Server: A3.VERISIGNDNS.COM
DNSSEC: unSigned
Registrar Abuse Contact Email: abuse@enom.com
Registrar Abuse Contact Phone: +1.4252982646
URL of the ICANN WHOIS Data Problem Reporting System: http://wdprs.internic.net/
>>> Last update of WHOIS database: 2017-01-13T05:52:52.00Z <<<

For more information on Whois status codes, please visit https://icann.org/epp


The data in this whois database is provided to you for information
purposes only, that is, to assist you in obtaining information about or
related to a domain name registration record. We make this information
available "as is," and do not guarantee its accuracy. By submitting a
whois query, you agree that you will use this data only for lawful
purposes and that, under no circumstances will you use this data to: (1)
enable high volume, automated, electronic processes that stress or load
this whois database system providing you this information; or (2) allow,
enable, or otherwise support the transmission of mass unsolicited,
commercial advertising or solicitations via direct mail, electronic
mail, or by telephone. The compilation, repackaging, dissemination or
other use of this data is expressly prohibited without prior written
consent from us.

We reserve the right to modify these terms at any time. By submitting
this query, you agree to abide by these terms.
Version 6.3 4/3/2002

Get Noticed on the Internet!  Increase visibility for this domain name by listing it at www.whoisbusinesslistings.com
Connecting to whois.enom.com...

WHOIS Server: whois.enom.com
Registrar URL: www.enom.com
Updated Date: 2017-01-13T05:52:52.00Z
Creation Date: 2000-08-11T12:15:25.00Z
Registrar Registration Expiration Date: 2025-08-11T16:15:00.00Z
Registrar: ENOM, INC.
Registrar IANA ID: 48
Reseller: NAMECHEAP, INC
Domain Status: clientTransferProhibited https://www.icann.org/epp#clientTransferProhibited
Registry Registrant ID:
Registrant Name: NAMECHEAP.COM NAMECHEAP.COM
Registrant Organization: NAMECHEAP, INC
Registrant Street: 4600 EAST WASHINGTON STREET, SUITE 305
Registrant City: PHOENIX
Registrant State/Province: AZ
Registrant Postal Code: 85034
Registrant Country: US
Registrant Phone: +1.6613102107
Registrant Phone Ext:
Registrant Fax: +1.6613102107
Registrant Fax Ext:
Registrant Email: SUPPORT@NAMECHEAP.COM
Registry Admin ID:
Admin Name: NAMECHEAP.COM NAMECHEAP.COM
Admin Organization: NAMECHEAP, INC
Admin Street: 4600 EAST WASHINGTON STREET, SUITE 305
Admin City: PHOENIX
Admin State/Province: AZ
Admin Postal Code: 85034
Admin Country: US
Admin Phone: +1.6613102107
Admin Phone Ext:
Admin Fax: +1.6613102107
Admin Fax Ext:
Admin Email: SUPPORT@NAMECHEAP.COM
Registry Tech ID:
Tech Name: NAMECHEAP.COM NAMECHEAP.COM
Tech Organization: NAMECHEAP, INC
Tech Street: 4600 EAST WASHINGTON STREET, SUITE 305
Tech City: PHOENIX
Tech State/Province: AZ
Tech Postal Code: 85034
Tech Country: US
Tech Phone: +1.6613102107
Tech Phone Ext:
Tech Fax: +1.6613102107
Tech Fax Ext:
Tech Email: SUPPORT@NAMECHEAP.COM
Name Server: A1.VERISIGNDNS.COM
Name Server: A2.VERISIGNDNS.COM
Name Server: A3.VERISIGNDNS.COM
DNSSEC: unSigned
Registrar Abuse Contact Email: abuse@enom.com
Registrar Abuse Contact Phone: +1.4252982646
URL of the ICANN WHOIS Data Problem Reporting System: http://wdprs.internic.net/
>>> Last update of WHOIS database: 2017-01-13T05:52:52.00Z <<<

For more information on Whois status codes, please visit https://icann.org/epp


The data in this whois database is provided to you for information
purposes only, that is, to assist you in obtaining information about or
related to a domain name registration record. We make this information
available "as is," and do not guarantee its accuracy. By submitting a
whois query, you agree that you will use this data only for lawful
purposes and that, under no circumstances will you use this data to: (1)
enable high volume, automated, electronic processes that stress or load
this whois database system providing you this information; or (2) allow,
enable, or otherwise support the transmission of mass unsolicited,
commercial advertising or solicitations via direct mail, electronic
mail, or by telephone. The compilation, repackaging, dissemination or
other use of this data is expressly prohibited without prior written
consent from us.

We reserve the right to modify these terms at any time. By submitting
this query, you agree to abide by these terms.
Version 6.3 4/3/2002

Get Noticed on the Internet!  Increase visibility for this domain name by listing it at www.whoisbusinesslistings.com
Connecting to whois.enom.com...

WHOIS Server: whois.enom.com
Registrar URL: www.enom.com
Updated Date: 2017-01-13T05:52:52.00Z
Creation Date: 2000-08-11T12:15:25.00Z
Registrar Registration Expiration Date: 2025-08-11T16:15:00.00Z
Registrar: ENOM, INC.
Registrar IANA ID: 48
Reseller: NAMECHEAP, INC
Domain Status: clientTransferProhibited https://www.icann.org/epp#clientTransferProhibited
Registry Registrant ID:
Registrant Name: NAMECHEAP.COM NAMECHEAP.COM
Registrant Organization: NAMECHEAP, INC
Registrant Street: 4600 EAST WASHINGTON STREET, SUITE 305
Registrant City: PHOENIX
Registrant State/Province: AZ
Registrant Postal Code: 85034
Registrant Country: US
Registrant Phone: +1.6613102107
Registrant Phone Ext:
Registrant Fax: +1.6613102107
Registrant Fax Ext:
Registrant Email: SUPPORT@NAMECHEAP.COM
Registry Admin ID:
Admin Name: NAMECHEAP.COM NAMECHEAP.COM
Admin Organization: NAMECHEAP, INC
Admin Street: 4600 EAST WASHINGTON STREET, SUITE 305
Admin City: PHOENIX
Admin State/Province: AZ
Admin Postal Code: 85034
Admin Country: US
Admin Phone: +1.6613102107
Admin Phone Ext:
Admin Fax: +1.6613102107
Admin Fax Ext:
Admin Email: SUPPORT@NAMECHEAP.COM
Registry Tech ID:
Tech Name: NAMECHEAP.COM NAMECHEAP.COM
Tech Organization: NAMECHEAP, INC
Tech Street: 4600 EAST WASHINGTON STREET, SUITE 305
Tech City: PHOENIX
Tech State/Province: AZ
Tech Postal Code: 85034
Tech Country: US
Tech Phone: +1.6613102107
Tech Phone Ext:
Tech Fax: +1.6613102107
Tech Fax Ext:
Tech Email: SUPPORT@NAMECHEAP.COM
Name Server: A1.VERISIGNDNS.COM
Name Server: A2.VERISIGNDNS.COM
Name Server: A3.VERISIGNDNS.COM
DNSSEC: unSigned
Registrar Abuse Contact Email: abuse@enom.com
Registrar Abuse Contact Phone: +1.4252982646
URL of the ICANN WHOIS Data Problem Reporting System: http://wdprs.internic.net/
>>> Last update of WHOIS database: 2017-01-13T05:52:52.00Z <<<

For more information on Whois status codes, please visit https://icann.org/epp


The data in this whois database is provided to you for information
purposes only, that is, to assist you in obtaining information about or
related to a domain name registration record. We make this information
available "as is," and do not guarantee its accuracy. By submitting a
whois query, you agree that you will use this data only for lawful
purposes and that, under no circumstances will you use this data to: (1)
enable high volume, automated, electronic processes that stress or load
this whois database system providing you this information; or (2) allow,
enable, or otherwise support the transmission of mass unsolicited,
commercial advertising or solicitations via direct mail, electronic
mail, or by telephone. The compilation, repackaging, dissemination or
other use of this data is expressly prohibited without prior written
consent from us.

We reserve the right to modify these terms at any time. By submitting
this query, you agree to abide by these terms.
Version 6.3 4/3/2002

Get Noticed on the Internet!  Increase visibility for this domain name by listing it at www.whoisbusinesslistings.com
Connecting to whois.enom.com...
^C
C:\Users\Brian G>

```

**Comment by BrianG61UK** [comment]
Hello? Is there anybody out there?

**Comment by wkl17** [comment]
I have the same problem with whois.exe v1.2 . Run on Win 10 x64 Professional (Build 10586).

**Comment by vfrazao-ns1** [comment]
Same issue here - also using v1.2.

**Comment by ashleedawg** [comment]
Same issue here - v1.2 on Windows 7

### **Workaround:**
"Endless loop" doesn't occur when using an older version of `whois.exe`, like v1.12.

- File is attached: [`whois_v112.zip`\*](https://github.com/MicrosoftDocs/sysinternals/files/2514590/whois_v112.zip) (Source is a link from [here](https://social.technet.microsoft.com/Forums/en-US/57386673-df81-40c6-9764-c2c7785b8207/whois-goes-into-an-endless-loop-with-godaddy-possibly-others?forum=miscutils)\*)  

<sup>\*_I make no claims as to the safety of this file, although using it didn't seem to cause a meltdown, yet._</sup>

**Comment by lukekim** [comment]
Thanks for reporting this, it should be fixed in v1.21.

**Comment by BrianG61UK** [comment]
Excellent.
Thank you.

**Comment by ashleedawg** [comment]
> 
> Thanks for reporting this, it should be fixed in v1.21.

When will that version become available?

**Comment by lukekim** [comment]
v1.21 has now been released: https://docs.microsoft.com/en-us/sysinternals/#whats-new-december-11-2019
